benchmarks/tpcc/src/main/java/com/google/cloud/pgadapter/tpcc/AbstractBenchmarkRunner.java

```diff
@@ -0,0 +1,781 @@
+    } else {
+      customerWarehouseId = getOtherWarehouseId(warehouseId);
+      customerDistrictId =
+          Long.reverse(random.nextInt(tpccConfiguration.getDistrictsPerWarehouse()));
+    }
+
+    execute(statement, "begin transaction");
+    executeParamStatement(
+        connection,
+        "UPDATE warehouse " + "SET w_ytd = w_ytd + ? " + "WHERE w_id = ?",
+        new Object[] {amount, warehouseId});
+
+    row =
+        paramQueryRow(
+            connection,
+            "SELECT w_street_1, w_street_2, w_city, w_state, w_zip, w_name "
+                + "FROM warehouse "
+                + "WHERE w_id = ?",
+            new Object[] {warehouseId});
+    String warehouseStreet1 = (String) row[0];
+    String warehouseStreet2 = (String) row[1];
+    String warehouseCity = (String) row[2];
+    String warehouseState = (String) row[3];
+    String warehouseZip = (String) row[4];
+    String warehouseName = (String) row[5];
+
+    executeParamStatement(
+        connection,
+        "UPDATE district " + "SET d_ytd = d_ytd + ? " + "WHERE w_id = ? AND d_id= ?",
+        new Object[] {amount, warehouseId, districtId});
+
+    row =
+        paramQueryRow(
+            connection,
+            "SELECT d_street_1, d_street_2, d_city, d_state, d_zip, d_name "
+                + "FROM district "
+                + "WHERE w_id = ? AND d_id = ?",
+            new Object[] {warehouseId, districtId});
+    String districtStreet1 = (String) row[0];
+    String districtStreet2 = (String) row[1];
+    String districtCity = (String) row[2];
+    String districtState = (String) row[3];
+    String districtZip = (String) row[4];
+    String districtName = (String) row[5];
+
+    if (byName) {
+      row =
+          paramQueryRow(
+              connection,
+              "SELECT count(c_id) namecnt "
+                  + "FROM customer "
+                  + "WHERE w_id = ? AND d_id= ? AND c_last=?",
+              new Object[] {customerWarehouseId, customerDistrictId, lastName});
+      int nameCount = (int) (long) row[0];
+      if (nameCount % 2 == 0) {
+        nameCount++;
+      }
+      try (PreparedStatement stmt =
+          connection.prepareStatement(
+              "SELECT c_id "
+                  + "FROM customer "
+                  + "WHERE w_id = ? AND d_id= ? AND c_last=? "
+                  + "ORDER BY c_first")) {
+        int index = 0;
+        stmt.setLong(++index, customerWarehouseId);
+        stmt.setLong(++index, customerDistrictId);
+        stmt.setString(++index, lastName);
+        try (ResultSet resultSet = executeParamQuery(stmt)) {
+          for (int counter = 0; counter < nameCount; counter++) {
+            if (resultSet.next()) {
+              customerId = resultSet.getLong(1);
+            }
+          }
+        }
+      }
+    }
+    row =
+        paramQueryRow(
+            connection,
+            (tpccConfiguration.isLockScannedRanges() ? "/*@ lock_scanned_ranges=exclusive */" : "")
```

Comment:
What is the purpose of `lock_scanned_ranges ` hint? I did not know we need to add it to all queries/DMLs. 

I moved the logic to `paramQueryRow`, `executeParamStatement`, and also a few places using `executeParamQuery`.

Reply:
The hint is a potential optimization in a use-case like this. Whether it actually improves throughput or not, is currently not clear to me, but in theory it might. Adding this hint makes Spanner behave more like a 'traditional' database in case of lock conflicts; instead of optimistically only taking a read-lock, it will instruct Spanner to take an exclusive write lock.

And now that I think of it: I was the one who only added this in some places.... I added it where the standard implementation would use `select ... for update`. Which is where we would potentially want to add this. Sorry for the confusion. Let's just leave it as is for now, and optimize this in a follow-up PR (if necessary).

Reply:
Yeah, probably, it's not quite right to use `exclusive lock` to every query and DML. I will create a PR to revert this change.